src/GoogleFit/FitOnFhir.GoogleFit/Services/GoogleFitDataImporter.cs

```diff
@@ -20,64 +19,47 @@ public class GoogleFitDataImporter : IGoogleFitDataImporter
         private readonly IGoogleFitClient _googleFitClient;
         private readonly IGoogleFitImportService _googleFitImportService;
         private readonly ILogger<GoogleFitDataImporter> _logger;
-        private readonly IUsersKeyVaultRepository _usersKeyvaultRepository;
-        private readonly IGoogleFitAuthService _googleFitAuthService;
+        private readonly IGoogleFitTokensService _googleFitTokensService;
         private readonly Func<DateTimeOffset> _utcNowFunc;
 
         public GoogleFitDataImporter(
             IUsersTableRepository usersTableRepository,
             IGoogleFitUserTableRepository googleFitUserTableRepository,
             IGoogleFitClient googleFitClient,
             IGoogleFitImportService googleFitImportService,
-            IUsersKeyVaultRepository usersKeyvaultRepository,
-            IGoogleFitAuthService googleFitAuthService,
+            IGoogleFitTokensService googleFitTokensService,
             Func<DateTimeOffset> utcNowFunc,
             ILogger<GoogleFitDataImporter> logger)
         {
             _usersTableRepository = EnsureArg.IsNotNull(usersTableRepository, nameof(usersTableRepository));
             _googleFitUserTableRepository = EnsureArg.IsNotNull(googleFitUserTableRepository, nameof(googleFitUserTableRepository));
             _googleFitClient = EnsureArg.IsNotNull(googleFitClient, nameof(googleFitClient));
             _googleFitImportService = EnsureArg.IsNotNull(googleFitImportService, nameof(googleFitImportService));
-            _usersKeyvaultRepository = EnsureArg.IsNotNull(usersKeyvaultRepository, nameof(usersKeyvaultRepository));
-            _googleFitAuthService = EnsureArg.IsNotNull(googleFitAuthService, nameof(googleFitAuthService));
+            _googleFitTokensService = EnsureArg.IsNotNull(googleFitTokensService, nameof(googleFitTokensService));
             _utcNowFunc = EnsureArg.IsNotNull(utcNowFunc);
             _logger = EnsureArg.IsNotNull(logger, nameof(logger));
         }
 
         /// <inheritdoc/>
         public async Task Import(string userId, string googleFitId, CancellationToken cancellationToken)
         {
-            string refreshToken;
-
-            _logger.LogInformation("Get RefreshToken from KV for {0}", googleFitId);
+            AuthTokensResponse tokensResponse;
 
             try
             {
-                refreshToken = await _usersKeyvaultRepository.GetByName(googleFitId, cancellationToken);
+                tokensResponse = await _googleFitTokensService.RefreshToken(googleFitId, cancellationToken);
             }
             catch (AggregateException ex)
```

Comment:
I don't think we should allow exceptions to bubble up to the caller. I think that the abstraction should handle exceptions and provide the caller with details of a failure, either through it's own exception type, our possibly a response object that contains details.